cmd/capslock-git-diff/main.go

```diff
@@ -0,0 +1,448 @@
+// Copyright 2024 Google LLC
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+// capslock-git-diff lists package capabilities that were added between
+// two revisions of a git repository.
+//
+// Usage example:
+//
+//	capslock-git-diff main mybranch somepath/...
+//
+// This requires Capslock to be installed:
+//
+//	go install github.com/google/capslock/cmd/capslock@latest
+//
+// To compare against the current state of the repository, specify "." as a
+// revision:
+//
+//	capslock-git-diff main . somepath/...
+//
+// If only two arguments are supplied, all packages under the current directory
+// are used.
+//
+// If the environment variable CAPSLOCKTOOLSTMPDIR is set and non-empty, it
+// specifies the directory where temporary files are created.  Otherwise the
+// system temporary directory is used.
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"slices"
+	"sort"
+	"strings"
+	"text/tabwriter"
+
+	cpb "github.com/google/capslock/proto"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+var (
+	verbose          = flag.Bool("v", false, "enable verbose logging")
+	granularity      = flag.String("granularity", "intermediate", "the granularity to use for comparisons")
+	flagCapabilities = flag.String("capabilities", "-UNANALYZED,-REFLECT,-UNSAFE_POINTER", "if non-empty, a comma-separated list of capabilities to pass to capslock")
+)
+
+func vlog(format string, a ...any) {
+	if !*verbose {
+		return
+	}
+	log.Printf(format, a...)
+}
+
+// run executes the specified command and writes its stdout to w.
+func run(w io.Writer, command string, args ...string) error {
+	vlog("running %q with args %q", command, args)
+	cmd := exec.Command(command, args...)
+	cmd.Stdout = w
+	if *verbose {
+		cmd.Stderr = os.Stderr
+	}
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("running %q with args %q: %w", command, args, err)
```

Comment:
It would be nice if the error was more informative, especially when running from a path with no Go files or with a branch name that doesn't exist. Maybe suggest using the -v flag to get more details on what went wrong?

Reply:
Added output of the command line and the command's standard error output when the command fails, even if -v is false.